### PR TITLE
Add Nasnet Implementation

### PR DESCRIPTION
**Pull request by hetong007**
WIP. Will merge with trained nasnet model.

### CODE REVIEW

**Comment by mli** [comment]
Job PR-136-5 is done. 
Docs are uploaded to http://gluon-vision-staging.s3-website-us-west-2.amazonaws.com/PR-136/5/index.html 
Code coverage of this PR:  vs. Master:

**Comment by mli** [comment]
Job PR-136-7 is done. 
Docs are uploaded to http://gluon-vision-staging.s3-website-us-west-2.amazonaws.com/PR-136/7/index.html 
Code coverage of this PR:  vs. Master:

**Comment by mli** [comment]
Job PR-136-10 is done. 
Docs are uploaded to http://gluon-vision-staging.s3-website-us-west-2.amazonaws.com/PR-136/10/index.html 
Code coverage of this PR:  vs. Master:

**Comment by mli** [comment]
Job PR-136-13 is done. 
Docs are uploaded to http://gluon-vision-staging.s3-website-us-west-2.amazonaws.com/PR-136/13/index.html 
Code coverage of this PR:  vs. Master:

**Comment by mli** [comment]
Job PR-136-14 is done. 
Docs are uploaded to http://gluon-vision-staging.s3-website-us-west-2.amazonaws.com/PR-136/14/index.html 
Code coverage of this PR:  vs. Master:

**Comment by mli** [comment]
Job PR-136-15 is done. 
Docs are uploaded to http://gluon-vision-staging.s3-website-us-west-2.amazonaws.com/PR-136/15/index.html 
Code coverage of this PR:  vs. Master:

**Comment by mli** [comment]
Job PR-136-16 is done. 
Docs are uploaded to http://gluon-vision-staging.s3-website-us-west-2.amazonaws.com/PR-136/16/index.html 
Code coverage of this PR:  vs. Master:

**Comment by mli** [comment]
Job PR-136-18 is done. 
Docs are uploaded to http://gluon-vision-staging.s3-website-us-west-2.amazonaws.com/PR-136/18/index.html 
Code coverage of this PR:  vs. Master:

**Comment by mli** [comment]
Job PR-136-19 is done. 
Docs are uploaded to http://gluon-vision-staging.s3-website-us-west-2.amazonaws.com/PR-136/19/index.html 
Code coverage of this PR:  vs. Master:

File: gluoncv/loss.py [hunk 1]
```diff
@@ -160,3 +160,87 @@ def forward(self, cls_pred, box_pred, cls_target, box_target):
...
+            loss = -F.pick(pred, label, axis=self._axis, keepdims=True)
+            loss = F.where(label.expand_dims(axis=self._axis) == self._ignore_label,
+                           F.zeros_like(loss), loss)
+        else:
+            label = _reshape_like(F, label, pred)
+            loss = -F.sum(pred*label, axis=self._axis, keepdims=True)
+        loss = _apply_weighting(F, loss, self._weight, sample_weight)
+        if self._size_average and self._sparse_label:
+            return F.mean(loss, axis=self._batch_axis, exclude=True) * \
+                valid_label_map.size / F.sum(valid_label_map)
```
Comment (zhanghang1989) [inline gluoncv/loss.py]: Actually, this hacky implementation can't be hybridized due to the `.size` method is used.

Comment (hetong007) [reply gluoncv/loss.py]: Good catch.

It's size should be the same as label. Can we have it as an input parameter?

Comment (zhanghang1989) [reply gluoncv/loss.py]: Good point. That should work. Could you make that change?

Comment (zhanghang1989) [reply gluoncv/loss.py]: Maybe as an optional input.

Comment (hetong007) [reply gluoncv/loss.py]: Sure.

**Comment by mli** [comment]
Job PR-136-21 is done. 
Docs are uploaded to http://gluon-vision-staging.s3-website-us-west-2.amazonaws.com/PR-136/21/index.html 
Code coverage of this PR:  vs. Master:

**Comment by mli** [comment]
Job PR-136-22 is done. 
Docs are uploaded to http://gluon-vision-staging.s3-website-us-west-2.amazonaws.com/PR-136/22/index.html 
Code coverage of this PR:  vs. Master:

**Comment by hetong007** [comment]
Holding. Will update and merge when https://github.com/apache/incubator-mxnet/pull/11021 is ready.

**Comment by mli** [comment]
Job PR-136-23 is done. 
Docs are uploaded to http://gluon-vision-staging.s3-website-us-west-2.amazonaws.com/PR-136/23/index.html 
Code coverage of this PR:  vs. Master:

**Comment by mli** [comment]
Job PR-136-24 is done. 
Docs are uploaded to http://gluon-vision-staging.s3-website-us-west-2.amazonaws.com/PR-136/24/index.html 
Code coverage of this PR:  vs. Master:

**Comment by mli** [comment]
Job PR-136-27 is done. 
Docs are uploaded to http://gluon-vision-staging.s3-website-us-west-2.amazonaws.com/PR-136/27/index.html 
Code coverage of this PR:  vs. Master:

File: gluoncv/loss.py [hunk 2]
```diff
@@ -160,3 +160,91 @@ def forward(self, cls_pred, box_pred, cls_target, box_target):
...
+            label = _reshape_like(F, label, pred)
+            loss = -F.sum(pred*label, axis=self._axis, keepdims=True)
+        loss = _apply_weighting(F, loss, self._weight, sample_weight)
+        if self._size_average and self._sparse_label:
+            return F.mean(loss, axis=self._batch_axis, exclude=True) * \
+                valid_size / F.sum(valid_label_map)
+        else:
+            return F.mean(loss, axis=self._batch_axis, exclude=True)
+
+class SoftmaxCrossEntropyLossWithAux(SoftmaxCrossEntropyLoss):
```
Comment (piiswrong) [inline gluoncv/loss.py]: Looks complicated. I don't think this is necessary. How about use two separate losses and add them in the training loop?

**Comment by mli** [comment]
Job PR-136-28 is done. 
Docs are uploaded to http://gluon-vision-staging.s3-website-us-west-2.amazonaws.com/PR-136/28/index.html 
Code coverage of this PR:  vs. Master:

**Comment by mli** [comment]
Job PR-136-31 is done. 
Docs are uploaded to http://gluon-vision-staging.s3-website-us-west-2.amazonaws.com/PR-136/31/index.html 
Code coverage of this PR:  vs. Master:

File: gluoncv/loss.py (hunk 2 above)
Comment (hetong007) [reply gluoncv/loss.py]: In segmentation it is also being used often. @zhanghang1989

**Comment by mli** [comment]
Job PR-136-49 is done. 
Docs are uploaded to http://gluon-vision-staging.s3-website-us-west-2.amazonaws.com/PR-136/49/index.html 
Code coverage of this PR:  vs. Master:

**Comment by mli** [comment]
Job PR-136-50 is done. 
Docs are uploaded to http://gluon-vision-staging.s3-website-us-west-2.amazonaws.com/PR-136/50/index.html 
Code coverage of this PR:  vs. Master:

**Comment by mli** [comment]
Job PR-136-53 is done. 
Docs are uploaded to http://gluon-vision-staging.s3-website-us-west-2.amazonaws.com/PR-136/53/index.html 
Code coverage of this PR:  vs. Master:
